sgeop/geometry.py

```diff
@@ -49,8 +42,17 @@ def _is_within(
     return np.abs(shapely.length(intersection) - shapely.length(line)) <= rtol
 
 
-def angle_between_two_lines(line1, line2):
-    # based on momepy.coins but adapted to shapely lines
+def angle_between_two_lines(
+    line1: shapely.LineString, line2: shapely.LineString
+) -> float:
+    """Return the angle between 2 two lines (assuming they share a vertex).
+    Based on ``momepy.coins`` but adapted to shapely lines.
+    """
+
+    lines_distinct = line1 != line2
+    if not lines_distinct:
+        raise ValueError("Input lines are identical - must be distinct.")
```

Comment:
Can we avoid adding any errors? I spent quite some time ensuring a single erroneous geometry does not bomb the whole algo on an error which indeed causes an issue in the simplification but only minor.  Can we warn and return something?

Reply:
That's probably reasonable. As for a returned value, which of the following would we think are the most realistic:
* `0.0`
* `numpy.inf`
* `numpy.nan`
* something else?

Reply:
I'd go with 0. It is a real value that may not break stuff down the line and the angle is 0 anyway.